TestModels/ExternV2/README.md

```diff
@@ -0,0 +1,41 @@
+# SimpleExternV2
+
+This project tests implementing a [dafny extern](https://homepage.cs.uiowa.edu/~tinelli/classes/181/Papers/dafny-reference.pdf#15) using the "V2 Extern" system.
+This models and its shapes are identical to operations in the `Extern` TestModel, only with "V2" suffixes appended.
+
+This V2 extern system uses the `replaceable` and `replaces` keywords in conjunction with the `outer-module` compile flag to allow a module to have idiomatic extern names in each target language.
+
+Under this V2 system, manually-written Dafny source code should mark `module`s that are intended to be `extern` as `replaceable`.
```

Comment:
I think we should clarify here that by `extern` we mean having the module under a specific path, and not have an extern implementation.

Reply:
How about this:

```
Under this V2 system, a Dafny developer should not mark a module as `extern` if the path to the extern module depends on language-specific features. (e.g. an `:extern "sample.namespace.MyExternModule"` makes sense for Java externs, but the `.`s will break Python externs.)
Instead of marking this module as `extern`, the developer should mark the module as `replaceable`.
Then, they should create a new module that `replaces` the `replaceable` module. The developer should add the `extern` attribute to this new module. This module's `extern` attribute will define a language-idiomatic path syntax. The developer is expected to create a new module that `replaces` the `replaceable` module for each path syntax.
```